### ISSUE TITLE
Error on initial loading

### ISSUE THREAD

**Issue by captain-lungo** [leoherzog/googlegroupadvancedsettingspanel#1]
`Error	
ReferenceError: AdminDirectory is not defined
getAllGroups	@ Code.gs:16`

**Comment by leoherzog** [comment]
Did you follow step 4 in the instructions? Under the **Services** section on the left, click the ➕ plus button and add the **Admin SDK API** and the **Groups Settings API**.

**Comment by captain-lungo** [comment]
Yes, you were right. I thought I did everything right but on checking it the **Admin SDK API** was missing.
Now it lists all groups correctly but after selecting one group the UI hangs on _updating_ and the log lists all group-atributes as **error**.

**getGroupSettings**
`25.04.2023, 10:11:31	Fehlerbehebung	{
  "whoCanAssignTopics": "NONE",
  "whoCanPostAnnouncements": "OWNERS_AND_MANAGERS",
  "allowWebPosting": "true",
  "customRolesEnabledForSettingsToBeMerged": "false",
  "whoCanMarkDuplicate": "NONE",
  "whoCanApproveMembers": "ALL_MANAGERS_CAN_APPROVE",
  "whoCanMakeTopicsSticky": "NONE",
  "archiveOnly": "false",
  "allowGoogleCommunication": "false",
  "whoCanUnmarkFavoriteReplyOnAnyTopic": "NONE",
  "defaultSender": "DEFAULT_SELF",
  "maxMessageBytes": 26214400,
  "whoCanDiscoverGroup": "ALL_IN_DOMAIN_CAN_DISCOVER",
  "whoCanAddReferences": "NONE",
  "whoCanPostMessage": "ANYONE_CAN_POST",
  "whoCanMoveTopicsOut": "OWNERS_AND_MANAGERS",
  "whoCanAssistContent": "NONE",
  "whoCanAdd": "ALL_MANAGERS_CAN_ADD",
  "whoCanModerateMembers": "OWNERS_AND_MANAGERS",
  "whoCanViewMembership": "ALL_IN_DOMAIN_CAN_VIEW",
  "whoCanViewGroup": "ALL_IN_DOMAIN_CAN_VIEW",
  "messageModerationLevel": "MODERATE_NONE",
  "whoCanDeleteTopics": "OWNERS_AND_MANAGERS",
  "whoCanModerateContent": "OWNERS_AND_MANAGERS",
  "messageDisplayFont": "DEFAULT_FONT",
  "whoCanDeleteAnyPost": "OWNERS_AND_MANAGERS",
  "customFooterText": "",
  "email": "test-group@some.domain",
  "whoCanModifyMembers": "OWNERS_AND_MANAGERS",
  "whoCanInvite": "ALL_MANAGERS_CAN_INVITE",
  "whoCanHideAbuse": "NONE",
  "whoCanEnterFreeFormTags": "NONE",
  "defaultMessageDenyNotificationText": "",
  "includeCustomFooter": "false",
  "whoCanMoveTopicsIn": "OWNERS_AND_MANAGERS",
  "whoCanJoin": "INVITED_CAN_JOIN",
  "spamModerationLevel": "MODERATE",
  "whoCanUnassignTopic": "NONE",
  "whoCanBanUsers": "OWNERS_AND_MANAGERS",
  "whoCanModifyTagsAndCategories": "NONE",
  "whoCanMarkNoResponseNeeded": "NONE",
  "whoCanMarkFavoriteReplyOnOwnTopic": "NONE",
  "includeInGlobalAddressList": "true",
  "isArchived": "true",
  "name": "Test-Group",
  "sendMessageDenyNotification": "false",
  "kind": "groupsSettings#groups",
  "whoCanContactOwner": "ANYONE_CAN_CONTACT",
  "whoCanTakeTopics": "NONE",
  "allowExternalMembers": "false",
  "membersCanPostAsTheGroup": "false",
  "enableCollaborativeInbox": "false",
  "whoCanApproveMessages": "OWNERS_AND_MANAGERS",
  "whoCanMarkFavoriteReplyOnAnyTopic": "NONE",
  "favoriteRepliesOnTop": "true",
  "whoCanLockTopics": "OWNERS_AND_MANAGERS",
  "replyTo": "REPLY_TO_IGNORE",
  "showInGroupDirectory": "true",
  "description": "Testgrppe",
  "whoCanLeaveGroup": "NONE_CAN_LEAVE"
}`
